crates/rooch-server/src/server/rooch_server.rs

```diff
@@ -93,34 +93,40 @@ impl RoochAPIServer for RoochServer {
         event_handle_type: StructTagView,
         cursor: Option<u64>,
         limit: Option<u64>,
-    ) -> RpcResult<Vec<Option<AnnotatedEventView>>> {
-        Ok(self
+    ) -> RpcResult<EventPage> {
+        // NOTE: fetch one more object to check if there is next page
+        let u_limit = limit.unwrap_or(MAX_RESULT_LIMIT);
+        let mut result: Vec<Option<AnnotatedEventView>> = self
             .rpc_service
-            .get_events_by_event_handle(
-                event_handle_type.into(),
-                cursor.unwrap_or(0),
-                limit.unwrap_or(MAX_RESULT_LIMIT),
-            )
+            .get_events_by_event_handle(event_handle_type.into(), cursor.unwrap_or(0), u_limit + 1)
             .await?
             .into_iter()
             .map(|event| event.map(AnnotatedEventView::from))
-            .collect())
+            .collect();
 
-        // let result: Vec<Option<AnnotatedEventView>> = Vec::new();
-        // Ok(result)
+        let has_next_page = (result.len() as u64) > u_limit;
+        let next_cursor = result.last().map_or(cursor, |event| {
+            Some(event.clone().unwrap().event.event_id.event_seq)
+        });
```

Comment:
Is the `cursor` included in the current page? so, the next_cursor should be last event_seq +1 ?

Reply:
every time fetch one more object to check if there is next page, so the last is just next cursor corresponding object